db.php

```diff
@@ -372,6 +372,12 @@ public function run_callbacks( $group, $args = null ) {
 		if ( ! isset( $args ) ) {
 			$args = array( &$this );
 		} elseif ( is_array( $args ) ) {
+			// 8.0+ changed the behavior of call_user_func_array(), associative arrays would turn into named attributes
+			// Here we discard the keys and hope for the best
+			if ( PHP_MAJOR_VERSION >= 8 ) {
+				$args = array_values( $args );
```

Comment:
Instead of leaving PHP 8-specific code in here, I think it would be better to just pass it through `array_values()` as-is since it shouldn't affect PHP 7.